File: c++/types/strings/search_substring.cpp
#include <iostream>
#include <typeinfo>

int main()
{
  std::string cite = "Stat rosa pristina nomine, nomina nuda tenemus.";
  std::cout << "cite = " << cite << std::endl << std::endl;

  int rosa_position = cite.find("rosa");
  std::cout << "\"rosa\" position: " << rosa_position << " (" << typeid(rosa_position).name() << ")\n";

  int odonata_position = cite.find("odonata");
  std::cout << "\"odonata\" position: " << odonata_position << " (" << typeid(odonata_position).name() << ")\n";
}

File: javascript/types/strings/search_substring.js
#!/usr/bin/env node

let cite = "Stat rosa pristina nomine, nomina nuda tenemus.";
console.log(`cite = ${cite}\n`);

let rosa_position = cite.indexOf("rosa");
console.log(`"rosa" position: ${rosa_position} (${typeof rosa_position})`);

let odonata_position = cite.indexOf("odonata");
console.log(`"odonata" position: ${odonata_position} (${typeof odonata_position})`);
